temporal-serviceclient/src/main/java/io/temporal/serviceclient/SimpleSslContextBuilder.java

```diff
@@ -47,21 +44,49 @@
           // gRPC requires http2 protocol.
           ApplicationProtocolNames.HTTP_2);
 
+  private final PKCS pkcs;
   private final InputStream keyCertChain;
   private final InputStream key;
   private TrustManager trustManager;
   private boolean useInsecureTrustManager;
   private String keyPassword;
 
+  private enum PKCS {
+    PKCS_8,
+    PKCS_12
+  }
+
   /**
    * @param keyCertChain - an input stream for an X.509 client certificate chain in PEM format.
    * @param key - an input stream for a PKCS#8 client private key in PEM format.
+   * @deprecated use {@link #forPKCS8(InputStream, InputStream)} isntead
    */
+  @Deprecated
   public static SimpleSslContextBuilder newBuilder(InputStream keyCertChain, InputStream key) {
-    return new SimpleSslContextBuilder(keyCertChain, key);
+    return forPKCS8(keyCertChain, key);
   }
 
-  private SimpleSslContextBuilder(InputStream keyCertChain, InputStream key) {
+  /**
+   * @param keyCertChain - an input stream for an X.509 client certificate chain in PEM format.
+   * @param key - an input stream for a PKCS#8 client private key in PEM format.
+   */
+  public static SimpleSslContextBuilder forPKCS8(InputStream keyCertChain, InputStream key) {
```

Comment:
It would have been nice if we could help users out and accept both pkcs8 and 12.